tests/suites/os/tests/secureboot/index.js

```diff
@@ -184,6 +173,113 @@ class uefiSecureBoot extends secureBoot {
 	};
 }
 
+class rpiSecureBoot extends secureBoot {
+	async isSecureBootSupported() {
+		return true;
+	}
+
+	async readPrivateKey() {
+		let reg = await this.worker.executeCommandInHostOS(
+			['vcmailbox', '0x00030081', '40 40 0 8 0 0 0 0 0 0 0 0'],
+			this.link
+		)
+		const key = reg
+			.replace(/0x/g, '') // remove 0x prefix
+			.trim()							// remove extra spaces
+			.split(/\s+/)				// split by whitespace
+			.slice(7, 15)				// take elements 8 to 15
+			.join('');          // concatenate
+		return key;
+	}
+
+	async readOTPReg(reg) {
+		let out = await this.worker.executeCommandInHostOS(
+			['vcgencmd', 'otp_dump', '2>/dev/null'],
+			this.link
+		)
+		for ( const line of out.split('\n')) {
+			if (line.startsWith(reg)) {
+				return line.split(':')[1].trim();
+			}
+		}
+	}
+
+	async checkRSAkey() {
+		for (let reg = 47; reg <= 54; reg++) {
+			let out = await this.readOTPReg(reg)
+			if ( out && out != "00000000") {
+				return true
+			}
+		}
+		return false
+	}
+
+	async isSecureBootEnabled() {
+		// OTP is programmed with customer key digest and private key
+		if ( await this.checkRSAkey() ) {
+			const key = await this.readPrivateKey();
+			if ( key.replace(/0/g, '').trim().length > 0 ) {
+				return true;
+			}
+		}
+	}
+
+	async waitForFailedBoot(){
+		// confirm that eth link is high
```

Comment:
@alexgg I haven't been able to test with these tests using the USB stick provisioning flow yet - so I haven't confirmed if this assertion is still valid. Does the device still fall back to network boot if it doesn't boot from emmc because of an invalid boot configuration etc?

Reply:
it still does @rcooke-warwick